### PR TITLE
GEOMESA-1476 Removing geohash index support

### PR DESCRIPTION
**Pull request by elahrvivaz**
Signed-off-by: Emilio Lahr-Vivaz elahrvivaz@ccri.com

### CODE REVIEW

File: geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/iterators/TestData.scala [hunk 1]
```diff
@@ -43,18 +41,6 @@ object TestData extends LazyLogging {
   val wktQuery = "POLYGON((45 23, 48 23, 48 27, 45 27, 45 23))"
 
   val featureName = "feature"
-  val schemaEncoding =
-    new IndexSchemaBuilder("~")
```
Comment (jahhulbert-ccri) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/iterators/TestData.scala]: awww i'll miss this guy

**Comment by elahrvivaz** [comment]
We still have the back-compatible tests going to 1.1.0-rc.7 - if you run them you can see the warning that gets logged, but how the tests still run using full table scans.
Good point about documentation - I'll add some.

File: geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/AccumuloFeatureIndex.scala [hunk 2]
```diff
@@ -100,16 +100,23 @@ object AccumuloFeatureIndex {
...
+    val indices = if (version > 8) {
+      CurrentIndices // note: version 9 was never in a release
+    } else if (version == 8) {
+      Seq(Z3IndexV2, Z2IndexV1, RecordIndexV1, AttributeIndexV2)
+    } else if (version > 5) {
+      logger.warn("The GeoHash index is no longer supported. Some queries make take longer than normal. To " +
+          s"update your data to a newer format, see $docs")
+      version match {
+        case 7 => Seq(Z3IndexV2, RecordIndexV1, AttributeIndexV2)
+        case 6 => Seq(Z3IndexV1, RecordIndexV1, AttributeIndexV2)
```
Comment (jnh5y) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/AccumuloFeatureIndex.scala]: I'm digging this way to sort things out.

**Comment by jnh5y** [comment]
Does this work with raster support;)?

**Comment by elahrvivaz** [comment]
also need to test out raster to ensure it still works

**Pull request merged**